### PR TITLE
[F] Add CloudAccountOrgSetupJob org mismatch case exception

### PR DESCRIPTION
**Pull request by nikosmoum**
- Defines adapter interface exception in case the cloud account setup cannot be done yet due to an org key conflict upstream of Candlepin.

### CODE REVIEW

**Comment by nikosmoum** [comment]
Note on sonar code coverage failure: lets ignore it because it's because of the new exception class

File: src/test/java/org/candlepin/async/tasks/CloudAccountOrgSetupJobTest.java [hunk 1]
```diff
@@ -187,6 +188,29 @@ void ensureJobExceptionThrownIfCouldNotEntitleOrganization()
...
+
+        CloudAccountOrgSetupJob regJob = new CloudAccountOrgSetupJob(cloudReg, ownerCurator);
+
+        CloudAccountOrgSetupJob.CloudAccountOrgSetupJobConfig jobConfig =
+            CloudAccountOrgSetupJob.createJobConfig()
+            .setCloudAccountId(TestUtil.randomString())
+            .setCloudOfferingId(TestUtil.randomString())
+            .setCloudProvider(CloudProvider.AZURE)
+            .setOwnerKey(TestUtil.randomString());
+
```
Comment (JoshAlbrecht23) [inline src/test/java/org/candlepin/async/tasks/CloudAccountOrgSetupJobTest.java:205]: Extra new line

**Review by JoshAlbrecht23 — CHANGES_REQUESTED** [review]
Looks good! I just noticed an extra new line

File: src/test/java/org/candlepin/async/tasks/CloudAccountOrgSetupJobTest.java (hunk 1 above)
Comment (nikosmoum) [reply src/test/java/org/candlepin/async/tasks/CloudAccountOrgSetupJobTest.java:205]: updated

**Review by JoshAlbrecht23 — APPROVED** [review]

**Pull request merged**